File: src/Application/Interfaces/Services/Module/Dto/ModuleDto.cs
namespace Application.Interfaces.Services.Module.Dto;

public class ModuleDto
{
    public string Id { get; set; } = null!;
    public string? Name { get; set; }
    public string? Subject { get; set; }
    public string? Content { get; set; }
    public string? CardImageUrl { get; set; }
    public List<ModuleSectionDto> Sections { get; set; } = new();
}

File: src/Web/vue-app/src/types/entities/moduleDto.ts
import type { ModuleSectionDto } from './moduleSection';

export interface ModuleDto {
    id: string;
    name?: string;
    nameFr?: string;
    nameEn?: string;
    subject?: string;
    sujetFr?: string;
    sujetEn?: string;
    content?: string;
    cardImageUrl?: string;
    isPublished?: boolean;
    sections?: ModuleSectionDto[];
}
